hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChunkBuffer.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.hadoop.ozone.common;
+
+import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.function.Function;
+
+/** Buffer for a block chunk. */
+public interface ChunkBuffer {
+  /** {@link ChunkBuffer} implementation using a single {@link ByteBuffer}. */
+  final class ImplWithByteBuffer implements ChunkBuffer {
```

Comment:
NIT: can we separate the ImplWithByteBuffer from ChunkBuffer interface in a separate file?

Reply:
Sure, just have pushed a new change to move out ImplWithByteBuffer.